pkg/util/util.go

```diff
@@ -33,3 +42,25 @@ func GetRequestHost(req *http.Request) string {
 	}
 	return host
 }
+
+// GetRequestPortFromHeader return the request port from X-Forwarded-Port if present, if not; use provided port
+func GetRequestPortFromHeader(req *http.Request, knownPort string) string {
+	port := req.Header.Get("X-Forwarded-Port")
+	if knownPort != "" {
+		port = knownPort
+	}
+	if port == "80" || port == "443" {
+		port = ""
+	}
```

Comment:
Why do we need this? If the port is specified in the request shouldn't we preserve it here? What's the intention of this? (This may well be needed but it's not obvious to me why right now, might just mean a comment is warranted)

Reply:
_If_ the port is standard, isValidRedirect gets tripped with "address / port not whitelisted" because host is now `https://a-service.example.com:443/` (notice the port)

Reply:
This probably can be solved by your previous suggestion, though. Will give it a try.

Reply:
Nope, apparently this has to stay.

Reply:
Ok, in that case could we add a small comment explaining why these are stripped above these lines?

Reply:
Will do!